### PR TITLE
NIP Proof of account

### PR DESCRIPTION
**Pull request by Dolu89**
Proof of account NIP

Draft, to discuss and validate

Discussion history : https://gist.github.com/Dolu89/1c8306004e6ad9b2fcbdcd1f87beb37e

### CODE REVIEW

**Comment by arcbtc** [comment]
I would like a proof where I create a bitcoin address from my handle and nostr public key, then burn some funds

and a proof where you post some message from twitter (seems this is already in the pipeline)

and a url proof

Using Lsat seems like a good idea as well https://twitter.com/KamilBrejcha/status/1358497277736742912

Will all these proofs have different prefixes?

File: nips/xx.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  - "t" : for twitter
+  - "g" : for github
+- `{url}` is the url of the proof 
+
+Example :
+`proof:{"pubkey":"19f....5dg", "type":"t", "url":"https://twitter.com/xxx/status/xxxxxxx"}`
+
+2. Relay store the proof. 
+
+3. Relay send back the the result to the client. Client has to validate the proof.
```
Comment (MaxHillebrand) [inline nips/xx.md:22]:
```suggestion
3. Relay send back the result to the client. Client has to validate the proof.
```

File: nips/xx.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+Proof of account protocol
+-------------------------------
+
+`draft` `mandatory`
+
+1. Users can prove they own some accounts by publishing an event to the relays
+`proof:<{pubkey}, {type}, {url}>`
+- `{pubkey}` is a pubkey, as a hex string
+- `{type}` is a string with a single letter, which can be:
```
Comment (MaxHillebrand) [inline nips/xx.md:12]: Are you sure you want to limit to 1 character identifiers? That's quite a small set size.

**Comment by MaxHillebrand** [comment]
> I would like a proof where I [...] burn some funds @arcbtc 

Instead of burning precious sats, you can lock them up in a fidelity bond. This has similar DoS protection yet doesn't decrease your total sat amount in the long run.

File: nips/xx.md (hunk 2 above)
Comment (Dolu89) [reply nips/xx.md:12]: You right, it's a bad idea. I think we should have a full name like "twitter" or "github"

**Comment by MaxHillebrand** [comment]
Maybe I'm missing something, but doesn't the protocol also need to specify the content of the tweet?

I mean, I can grab a tweet link from any tweet from anyone, and just include this in the proper format of the event.

I guess the easiest would be, to require that the tweet content is the nostr pubkey. It could be a nonce referenced in both tweet and event too.

**Comment by Giszmo** [comment]
nips moved to https://github.com/nostr-protocol/nips
We should close this and all other nip PRs.

This is currently not subject to the other repo.

**Pull request closed without merging**